### PR TITLE
App forensics/bulk_extractor updates

### PR DESCRIPTION
**Pull request by linxon**

### CODE REVIEW

**Comment by blshkv** [comment]
Please push these 2 patches to the upstream. You dont want to use a snapshot with custom (untrusted) patches in forensics works

File: app-forensics/bulk_extractor/bulk_extractor-1.5.5-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+src_prepare() {
+	# Add support hashdb-3.1.0 for old bulk_extractor versions
+	# https://github.com/NPS-DEEP/hashdb/wiki/hashdb-3.1.0-and-bulk_extractor
+	use hashdb && \
+		eapply "${FILESDIR}/${P}_hashdb-3.1.0_and_old_bulk_extractor.patch"
+
+	# Using -I rather than -isystem for BOOST_CPPFLAGS
+	# https://gcc.gnu.org/bugzilla/show_bug.cgi?id=70129
+	# Add exiv-0.27.0 support and other minor fixes...
+	eapply "${FILESDIR}/add_exiv2-0.27.0_support.patch"
```
Comment (linxon) [inline app-forensics/bulk_extractor/bulk_extractor-1.5.5-r1.ebuild]: i'm trying to run command as: 
```shell
linxon@cirno-chan ~/GitClones/pentoo-overlay $ bulk_extractor -R -e exiv2 ~/NewM/ -o ~/WorkDir/
Restarting from /home/linxon/WorkDir/
bulk_extractor version: 1.6.0-dev
Hostname: cirno-chan
Input file: /home/linxon/NewM/
Output directory: /home/linxon/WorkDir/
Disk Size: 141
Threads: 8
...
...
...
linxon@cirno-chan ~/GitClones/pentoo-overlay $ echo $?
0
```

File: app-forensics/bulk_extractor/files/add_exiv2-0.27.0_support.patch [hunk 2]
```diff
@@ -0,0 +1,14 @@
+diff -ur a/src/dfxml/src/dfxml_writer.cpp b/src/dfxml/src/dfxml_writer.cpp
+--- a/src/dfxml/src/dfxml_writer.cpp    2014-09-16 22:34:02.000000000 +0400
++++ b/src/dfxml/src/dfxml_writer.cpp    2019-03-17 10:28:31.797180265 +0300
+@@ -646,6 +646,10 @@
+ #include <exiv2/image.hpp>
+ #include <exiv2/exif.hpp>
+ #include <exiv2/error.hpp>
++#include <exiv2/exiv2.hpp>
++#if EXIV2_TEST_VERSION(0, 27, 0)
```
Comment (linxon) [inline app-forensics/bulk_extractor/files/add_exiv2-0.27.0_support.patch]:
```c++
#if EXIV2_TEST_VERSION(0, 27, 0)
...
#endif
```
EXIV2_TEST_VERSION can be defined after include the exiv2/exiv2.hpp header

File: app-forensics/bulk_extractor/files/fix_call_of_overloaded_errors.patch [hunk 3]
```diff
@@ -0,0 +1,24 @@
+diff -ur a/src/scan_exiv2.cpp b/src/scan_exiv2.cpp
+--- a/src/scan_exiv2.cpp	2014-09-16 22:34:00.000000000 +0400
++++ b/src/scan_exiv2.cpp	2019-03-17 08:38:29.479753464 +0300
+@@ -68,7 +68,7 @@
+  * Used for helping to convert libexiv2's GPS format to decimal lat/long
+  */
+ 
+-static double stod(string s)
++static double sub_stod(string s)
```
Comment (linxon) [inline app-forensics/bulk_extractor/files/fix_call_of_overloaded_errors.patch]: my `c++` knowledge is not enough here, but if you rename this function, the compiler will not report errors.
^_^

**Comment by blshkv** [comment]
ahha! Thanks for the exploitation. Yet, hashdb guys should push their patches back to bulk_extractor. Let's push it to Pentoo for testing

**Comment by blshkv** [comment]
for records,
https://github.com/simsong/bulk_extractor/issues/89

**Pull request merged**